src/creation/mod.rs

```diff
@@ -106,12 +106,15 @@ pub async fn run<H: Hasher>(
         mut incoming_parents,
         outgoing_units,
     } = io;
-    let starting_round = match starting_round.await {
-        Ok(round) => round,
-        Err(e) => {
-            error!(target: "AlephBFT-creator", "Starting round not provided: {}", e);
-            return;
-        }
+    let starting_round = futures::select! {
+        round =  starting_round => match round {
+            Ok(round) => round,
```

Comment:
Nit: These is a bit tough to read at first due to multiple occurences of `round` in this select block. I'd personally rename one of these, but up to you.